src/sentry/static/sentry/app/views/organizationDiscover/result/vizualizationsDropdown.jsx

```diff
@@ -0,0 +1,79 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import DropdownLink from 'app/components/dropdownLink';
+import MenuItem from 'app/components/menuItem';
+import {t} from 'app/locale';
+import styled from 'react-emotion';
+import theme from 'app/utils/theme';
+
+class VisualizationsDropdown extends React.PureComponent {
+  static propTypes = {
+    options: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string,
+        name: PropTypes.string,
+      })
+    ).isRequired,
+    handleChange: PropTypes.func.isRequired,
+    visualization: PropTypes.string.isRequired,
+  };
+
+  getMenuItem = opt => {
+    return (
+      <MenuItem
+        key={opt.id}
+        onSelect={this.props.handleChange}
+        eventKey={opt.id}
+        isActive={opt.id === this.props.visualization}
+      >
+        {opt.name}
+      </MenuItem>
+    );
+  };
+
+  getLabel = key => {
```

Comment:
Is there a way we can avoid defining the mappings between the visualization ids and display names a second time?

Reply:
I could map through the array and check if the id matches, is that preferable? I thought this would be a bit cleaner